Framework/DataHandling/src/LoadILLDiffraction.cpp

```diff
@@ -496,18 +499,26 @@ void LoadILLDiffraction::fillStaticInstrumentScan(const NXUInt &data, const NXDo
   const std::vector<double> axis = getAxis(scan);
   const std::vector<double> monitor = getMonitor(scan);
 
+  size_t monitorIndex = 0;
+  size_t startIndex = NUMBER_MONITORS;
+  if (m_instName == "IN5" || m_instName == "PANTHER" || m_instName == "SHARP") {
```

Comment:
This condition is checked twice, maybe define a variable `isSpectrometer` and reuse it.